WinUIGallery/ControlPages/FilePickerPage.xaml.cs

```diff
@@ -0,0 +1,276 @@
+            WinRT.Interop.InitializeWithWindow.Initialize(openPicker, hWnd);
+
+            // Set options for your folder picker
+            openPicker.SuggestedStartLocation = PickerLocationId.Desktop;
+            openPicker.FileTypeFilter.Add("*");
+
+            // Open the picker for the user to pick a folder
+            StorageFolder folder = await openPicker.PickSingleFolderAsync();
+            if (folder != null)
+            {
+                StorageApplicationPermissions.FutureAccessList.AddOrReplace("PickedFolderToken", folder);
+
+                // Creating the text for the Texblock
+                Span span = new Span();
+                Run run1 = new Run();
+                run1.Text = "Picked folder: ";
+
+                // Adding the name of the picked file in bold
+                Run run2 = new Run();
+                run2.FontWeight = Microsoft.UI.Text.FontWeights.Bold;
+                run2.Text = folder.Name;
+
+                span.Inlines.Add(run1);
+                span.Inlines.Add(run2);
+                PickFolderOutputTextBlock.Inlines.Add(span);
+            }
+            else
+            {
+                PickFolderOutputTextBlock.Text = "Operation cancelled.";
+            }
+        }
+
+        private async void SaveFileButton_Click(object sender, RoutedEventArgs e)
+        {
+            // Clear previous returned file name, if it exists, between iterations of this scenario
+            SaveFileOutputTextBlock.Text = "";
+
+            // Create a file picker
+            FileSavePicker savePicker = new Windows.Storage.Pickers.FileSavePicker();
+
+            // Retrieve the window handle (HWND) of the current WinUI 3 window.
+            var window = WindowHelper.GetWindowForElement(this);
+            var hWnd = WinRT.Interop.WindowNative.GetWindowHandle(window);
+
+            // Initialize the file picker with the window handle (HWND).
+            WinRT.Interop.InitializeWithWindow.Initialize(savePicker, hWnd);
+
+            // Set options for your file picker
+            savePicker.SuggestedStartLocation = PickerLocationId.DocumentsLibrary;
+            // Dropdown of file types the user can save the file as
+            savePicker.FileTypeChoices.Add("Plain Text", new List<string>() { ".txt" });
+            // Default file name if the user does not type one in or select a file to replace
+            var enteredFileName = ((sender as Button).Parent as StackPanel)
+            .FindName("FileNameTextBox") as TextBox;
+            savePicker.SuggestedFileName = enteredFileName.Text;
+
+            // Open the picker for the user to pick a file
+            StorageFile file= await savePicker.PickSaveFileAsync();
+            if (file != null)
+            {
+                // Prevent updates to the remote version of the file until we finish making changes and call CompleteUpdatesAsync.
+                CachedFileManager.DeferUpdates(file);
```

Comment:
This line seems to crash on my machine, @gregwoo-microsoft does this also happen on your machine?

Reply:
This works fine for me. What kind of error are you getting?

Reply:
I'm getting this exception:

![Screenshot of InvalidCastException](https://user-images.githubusercontent.com/16122379/211660329-3864750e-c5b3-40dd-a0bb-9d3d3f4e3ebc.png)